orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java

```diff
@@ -62,4 +62,17 @@ public class OrchestrationModuleConfig {
   public OrchestrationModuleConfig withLlmConfig(@Nonnull final OrchestrationAiModel aiModel) {
     return withLlmConfig(aiModel.createConfig());
   }
+
+  /**
+   * Creates a new configuration with the given Data Masking configuration.
+   *
+   * @param dpiMasking The Data Masking configuration to use.
+   * @return A new configuration with the given Data Masking configuration.
+   */
+  @Tolerate
+  @Nonnull
+  public OrchestrationModuleConfig withDpiMaskingConfig(
+      @Nonnull final DpiMaskingConfig dpiMasking) {
```

Comment:
Masking has a similar challenge like filtering: multiple masking provider configs are allowed, and they can be from different providers. 

Here, we always replace the entire masking module config, which is t ideal. Maybe compare with how the filtering PR solves the problem

Reply:
Ok, then I will not call it `with` but `add`

Reply:
`add` has the problem that it sounds like we are mutating, but our object is designed to be immutable